### PR TITLE
Two Asset Linearized Time Series Feed

### PR DESCRIPTION
**Pull request by richardliang**

### CODE REVIEW

File: contracts/meta-oracles/feeds/TwoAssetLinearizedTimeSeriesFeed.sol [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+     * owner is allowed to change.
+     *
+     * @param  _newBaseOracleAddress       Address of new oracle to pull data from
+     */
+    function changeBaseOracle(
+        IOracle _newBaseOracleAddress
+    )
+        external
+        onlyOwner
+        timeLockUpgrade // Must be placed after onlyOwner
```
Comment (felix2feng) [inline contracts/meta-oracles/feeds/TwoAssetLinearizedTimeSeriesFeed.sol]: Let's use `timeLockUpgradeV2` from now on. I'm exporting it from the latest version of `set-protocol-contracts:1.2.9`.

It'll automatically be only callable by the owner

File: contracts/meta-oracles/oracles/LastPriceOracle.sol [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+pragma solidity 0.5.7;
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { IOracle } from "../interfaces/IOracle.sol";
+import { ITimeSeriesFeed } from "../interfaces/ITimeSeriesFeed.sol";
+
+
+/**
+ * @title LastPriceOracle
```
Comment (felix2feng) [inline contracts/meta-oracles/oracles/LastPriceOracle.sol]: Name suggestion: `LastValueOracle`

File: contracts/meta-oracles/feeds/TwoAssetLinearizedTimeSeriesFeed.sol [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+     * the previous value and the time that has elapsed since the intended update value.
+     *
+     * Returns with newest data point by querying oracle. Is eligible to be
+     * called after nextAvailableUpdate timestamp has passed. Because the nextAvailableUpdate occurs
+     * on a predetermined cadence based on the time of deployment, delays in calling poke do not propogate
+     * throughout the whole dataset and the drift caused by previous poke transactions not being mined
+     * exactly on nextAvailableUpdate do not compound as they would if it was required that poke is called
+     * an updateInterval amount of time after the last poke.
+     *
+     * @param  _timeSeriesState         Struct of TimeSeriesFeed state
```
Comment (bweick) [inline contracts/meta-oracles/feeds/TwoAssetLinearizedTimeSeriesFeed.sol]: Imagine this is a relic from another contract but there's no params

File: contracts/meta-oracles/oracles/LastValueOracle.sol [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    /* ============ State Variables ============ */
+    ITimeSeriesFeed public timeSeriesFeedInstance;
+    string public dataDescription;
+
+    /* ============ Constructor ============ */
+    /*
+     * Set price oracle is made to return
+     *
+     * @param  _timeSeriesFeedInstance    The address of base asset price feed
```
Comment (bweick) [inline contracts/meta-oracles/oracles/LastValueOracle.sol]: dataDescription

File: contracts/meta-oracles/feeds/TwoAssetLinearizedTimeSeriesFeed.sol [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+    /* ============ External ============ */
+
+    /*
+     * Change base asset oracle in case current one fails or is deprecated. Only contract
+     * owner is allowed to change.
+     *
+     * @param  _newBaseOracleAddress       Address of new oracle to pull data from
+     */
+    function changeBaseOracle(
```
Comment (bweick) [inline contracts/meta-oracles/feeds/TwoAssetLinearizedTimeSeriesFeed.sol]: Just had a thought...given there's a timeLock on here lets say we accidentally pass a non-oracle address then this whole thing would be messed up for potentially days. Does it make sense to just read a data point from the Oracle and make sure it doesn't revert?

Comment (bweick) [reply contracts/meta-oracles/feeds/TwoAssetLinearizedTimeSeriesFeed.sol]: Obviously same for the QuoteOracle as well

**Pull request merged**